jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/HttpException.java

```diff
@@ -0,0 +1,128 @@
+//
+// ========================================================================
+// Copyright (c) 1995 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.http;
+
+import org.eclipse.jetty.io.QuietException;
+import org.eclipse.jetty.util.ExceptionUtil;
+
+/**
+ * <p>A tagging interface for Exceptions that carry a HTTP response code and reason.</p>
+ * <p>Exception sub-classes that implement this interface will be caught by the container
+ * and the {@link #getCode()} used to send a response.</p>
+ */
+@SuppressWarnings("serial")
+public interface HttpException extends QuietException
+{
+    int getCode();
+
+    String getReason();
+
+    static Throwable asThrowable(HttpException httpException)
+    {
+        return httpException instanceof Throwable throwable ? throwable : null;
+    }
```

Comment:
I would remove this and replace its usages with a plain cast to `Throwable`, which fails if we mess up.